packages/dds/map/src/test/directory.spec.ts

```diff
@@ -136,6 +137,16 @@ describe("Directory", () => {
                 assert.equal(valueChangedExpected, false, "missing valueChangedExpected event");
                 assert.equal(containedValueChangedExpected, false, "missing containedValueChanged event");
 
+                // Test dispose on subdirectory delete
```

Comment:
It would be great to add steps in this test that would (at the end of current test)
- create sub-directory with same name 
- Validate that newly created object works and one can set / reset keys on it and it's not disposed
- The old in-memory object stays disposed, and accessing any key on it throws exception

Reply:
Can not add this scenario to test currently as with current implementation we return the same object on directory creation with same name. So previous one will not be disposed. This PR just add dispose event on explicitly deleting the directory.

Reply:
I think this is a bug. Once sub-directory is disposed, it should not be accessible from directory itself - it should be gone completely, and garbage-collected once users stop referencing it.

Reply:
Sorry I got the test wrong what you were saying. added the mentioned suggested test.